### PR TITLE
Missing Maps: improve visibility of "Calc online" and "Use existing maps sections", avoid double calc-online call

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculationResult.java [hunk 1]
```diff
@@ -14,6 +14,7 @@ public class MissingMapsCalculationResult {
 
 	private final RoutingContext missingMapsRoutingContext;
 	private final List<LatLon> missingMapsPoints;
+	private boolean onlineResultPoints = false;
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculationResult.java:17]: This had bad name & bad intention, I would like to understand the gist of this field and why it's necessary here

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java [hunk 2]
```diff
@@ -158,13 +159,14 @@ public boolean checkIfThereAreMissingMaps(RoutingContext ctx, LatLon start, List
 				}
 			}
 		}
-		
-//		System.out.println("Used maps: " + usedMaps);
+
+		if (result.hasMissingMaps() || calledByCalcOnline) {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java:163]: This is confusing as well

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculationResult.java (hunk 1 above)
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculationResult.java]: refactored

File: OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java (hunk 2 above)
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/router/MissingMapsCalculator.java]: refactored

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd/issues/19982

File: OsmAnd/src/net/osmand/plus/routepreparationmenu/RequiredMapsController.java [hunk 3]
```diff
@@ -222,4 +218,14 @@ public static void showDialog(@NonNull FragmentActivity activity) {
...
 	}
+
+	public boolean shouldShowOnlineCalculation() {
+		return !onlineCalculationRequested && !isLoadingInProgress() && !isAlreadyCalcOnline();
+	}
+
+	private boolean isAlreadyCalcOnline() {
+		RoutingHelper helper = app.getRoutingHelper();
+		MissingMapsCalculationResult result = helper.getRoute().getMissingMapsCalculationResult();
+		return result != null && result.getMissingMapsPoints().size() > 2 + helper.getIntermediatePoints().size();
```
Comment (alex-osm) [inline OsmAnd/src/net/osmand/plus/routepreparationmenu/RequiredMapsController.java:229]: Not a direct sign of calculation type - online or not. 
Perhaps a `String tag` field could be added to `MissingMapsCalculator` and `MissingMapsCalculationResult` to differentiate between types.

Comment (RZR-UA) [reply OsmAnd/src/net/osmand/plus/routepreparationmenu/RequiredMapsController.java:229]: This is the most obvious method to detect the fact that Calc Online is successfully done already.

We have a number of `missingMapsPoints` and a number of original routing points (`start+end+intermediates = 2 + intermediates`).

Before Calc Online, missingMapsPoints are the same as start+end+intermediates. After, it have a lot of calculated points (hunderds-thousands). Just compare!

The method is based on data that is already exist and, in my opinion, in this particular case we should not expand classes with new useless members just to copy-paste already-existing information.

Comment (RZR-UA) [reply OsmAnd/src/net/osmand/plus/routepreparationmenu/RequiredMapsController.java]: refactores to strange tag

**Review by alex-osm — APPROVED** [review]

**Comment by vshcherb** [comment]
- Separate requests
- Don't hide button + no changes in missing maps

**Pull request closed without merging**